FreeRTOS-Plus/Demo/coreSNTP_Windows_Simulator/DemoTasks/SampleAppTask.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * FreeRTOS V202104.00
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/*
+ * This file belongs to the demo for showing use of the coreSNTP library for synchronizing
+ * system time with internet time and maintaining correct wall-clock time.
+ *
+ * This file shows how an application task can query Coordinated Universal Time (UTC) from system,
+ * that is maintained and synchronized periodically in the SNTP client (daemon) task of the demo
+ * project. Refer to the the SNTPClientTask.c file in this project for the SNTP client task.
+ */
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <time.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Demo includes. */
+#include "common_demo_include.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The periodicity of the application task in query and logging system time. This is the time
+ * interval between consecutive system clock time queries in the sample application task.
+ */
+#define CLOCK_QUERY_TASK_DELAY_MS    ( 1000 )
+
+/**
+ * @brief The constant for storing the number of milliseconds per FreeRTOS tick in the system.
+ * @note This value represents the time duration per tick from the perspective of the
+ * of Windows Simulator based FreeRTOS system that carries lagging clock drift in relation to
+ * internet time or UTC time. Thus, the actual time duration value per tick of the system will be
+ * larger from the perspective of internet time.
+ */
+#define MILLISECONDS_PER_TICK        ( 1000 / configTICK_RATE_HZ )
```

Comment:
Seems unused.

Reply:
It is used in the `calculateCurrentTime` function to convert tick counts to milliseconds